api/integration/kubernetes/install/app_test.go

```diff
@@ -14,387 +12,16 @@ import (
-							Value:   multitype.FromString("value"),
-						},
-						{
-							Name:     "file-item",
-							Type:     "file",
-							Title:    "File Item",
-							Filename: "file.txt",
-							Default:  multitype.FromString("SGVsbG8="),
-							Value:    multitype.FromString("QQ=="),
-						},
-					},
-				},
-			},
-		},
-	}
-
-	// Create config values that should be applied to the config
-	configValues := types.AppConfigValues{
-		"test-item": types.AppConfigValue{Value: "applied-value"},
-		"file-item": types.AppConfigValue{Value: "SGVsbG8gV29ybGQ=", Filename: "new-file.txt"},
-	}
-
-	// Create an install controller with the config values
-	installController, err := kubernetesinstall.NewInstallController(
-		kubernetesinstall.WithConfigValues(configValues),
-		kubernetesinstall.WithReleaseData(&release.ReleaseData{
-			AppConfig: &appConfig,
-		}),
-	)
-	require.NoError(t, err)
-
-	// Create the API with the install controller
-	apiInstance := integration.NewAPIWithReleaseData(t,
-		api.WithKubernetesInstallController(installController),
-		api.WithAuthController(auth.NewStaticAuthController("TOKEN")),
-		api.WithLogger(logger.NewDiscardLogger()),
-	)
-
-	// Create a router and register the API routes
-	router := mux.NewRouter()
-	apiInstance.RegisterRoutes(router)
-
-	// Test successful get
-	t.Run("Success", func(t *testing.T) {
-		// Create a request
-		req := httptest.NewRequest(http.MethodGet, "/kubernetes/install/app/config/values", nil)
-		req.Header.Set("Authorization", "Bearer "+"TOKEN")
-		rec := httptest.NewRecorder()
-
-		// Serve the request
-		router.ServeHTTP(rec, req)
-
-		// Check the response
-		assert.Equal(t, http.StatusOK, rec.Code)
-		assert.Equal(t, "application/json", rec.Header().Get("Content-Type"))
-
-		// Parse the response body
-		var response types.AppConfigValuesResponse
-		err = json.NewDecoder(rec.Body).Decode(&response)
-		require.NoError(t, err)
-
-		// Verify the app config values are returned from the store
-		assert.Equal(t, configValues, response.Values, "app config values should be returned from store")
-	})
-
-	// Test authorization
-	t.Run("Authorization error", func(t *testing.T) {
-		// Create a request
-		req := httptest.NewRequest(http.MethodGet, "/kubernetes/install/app/config/values", nil)
-		req.Header.Set("Authorization", "Bearer "+"NOT_A_TOKEN")
-		rec := httptest.NewRecorder()
-
-		// Serve the request
-		router.ServeHTTP(rec, req)
-
-		// Check the response
-		assert.Equal(t, http.StatusUnauthorized, rec.Code)
-		assert.Equal(t, "application/json", rec.Header().Get("Content-Type"))
-
-		// Parse the response body
-		var apiError types.APIError
-		err = json.NewDecoder(rec.Body).Decode(&apiError)
-		require.NoError(t, err)
-		assert.Equal(t, http.StatusUnauthorized, apiError.StatusCode)
-	})
-}
-
 // TestInstallController_PatchAppConfigValuesWithAPIClient tests the PatchAppConfigValues endpoint using the API client
 func TestInstallController_PatchAppConfigValuesWithAPIClient(t *testing.T) {
```

Comment:
So, the test suite works well for our API tests, but things get more complicated with the API client ones and I hadn't considered that.

I don't think there's an easy/pretty way out this that doesn't involve passing around function parameters or instead creating an interface for the client (or changing the api for the client altogether). So, I'm curious for:
- Your thoughts?
- What do you think about doing these as follow ups?

Reply:
yeah, what you have in this pr is a great improvement, we can revisit these if they become a burden